android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -112,6 +112,36 @@ private void createChannel(ReactApplicationContext context) {
         mNotificationManager.createNotificationChannel(mChannel);
     }
 
+    private boolean hasControl(long control) {
+        if((controls & control) == control) {
+            return true;
+        }
+        return false;
+    }
+
+    private void updateNotificationMediaStyle() {
+        if (!(Build.MANUFACTURER.toLowerCase(Locale.getDefault()).contains("huawei") && Build.VERSION.SDK_INT < Build.VERSION_CODES.M)) {
+            MediaStyle style = new MediaStyle();
+            style.setMediaSession(session.getSessionToken());
+            int controlCount = 0;
+            if(hasControl(PlaybackStateCompat.ACTION_PLAY) || hasControl(PlaybackStateCompat.ACTION_PAUSE) || hasControl(PlaybackStateCompat.ACTION_PLAY_PAUSE)) {
+                controlCount += 1;
+            }
+            if(hasControl(PlaybackStateCompat.ACTION_SKIP_TO_NEXT)) {
+                controlCount += 1;
+            }
+            if(hasControl(PlaybackStateCompat.ACTION_SKIP_TO_PREVIOUS)) {
+                controlCount += 1;
+            }
+            int[] actions = new int[controlCount];
```

Comment:
Why not use an list instead and convert it to an array after?

Reply:
Because why bother...? It's an unnecessary extra step, and I think they were just asking you to check if the commit worked.

Reply:
Also they're still indices so you'd still need to increment controlCount, so you really wouldn't be trimming down that much code, if any.

Reply:
The code should be moved to `prepareNotification` from `MusicControlNotification`.
After adding the actions you can reduce the code like this:
```java
int[] actions = new int[builder.mActions.size()];
for(int i = 0; i < actions.length; i++) {
    actions[i] = i;
}
```

Reply:
okay.
well make sure it maxes out at 3, because the lock screen notification gets really weird when there's more

Reply:
I'm confused. Are you asking me to change this or are you changing this yourself?